### ISSUE TITLE
cbp6_link in dashboard code -- modularize!

### ISSUE THREAD

**Issue by rburghol** [HARPgroup/vahydro#45]
The dashboard scripts are now sophisticated enough that we will want to modularize these in R, so that functions can be called without hard-wiring a directory -- see the cbp6_link variable in https://github.com/HARPgroup/cbp6/blob/master/code/Working_Gage_Vs_Vahydro_Dashboard_2019.Rmd 

This is a more complex deal, and we need to decide if these functions should live in [cbp6 ](https://github.com/HARPgroup/cbp6) or here in vahydro.  My sense is that this is a bit of a judgement call, since some of our dashboards are built to compare vahydro things, and some are usd for cbp model analysis, and really, the goal was to build these to be agnostic, that is, they should just accept a couple time series data frames and go on their merry way -- I think we pretty much reached that goal, so again, location may be arbitrary for these base dashboards, though when we get to the SWRP analyses in the next few weeks we should definitely put them here in VAHydro.

**Comment by hdaniel7** [comment]
I agree that modularization is a great thing to pursue!  The one thing I have to point out is that our REST authentification information is stored within the auth.private file -- the only way I've found to load this file is by hard-wiring the directory containing this file.  Do you have any ideas on alternate ways to load these authentication variables which would not require hard-wiring?

**Comment by rburghol** [comment]
Good point on REST auth, I guess that needs to remain hard wired, unless we pass that file location in via a command line argument? Or alternatively, prompt the user to enter the auth data when running?

**Comment by hdaniel7** [comment]
The dashboard-generating Rmarkdown file has been updated to be modularized and to be able to take in data from any source.  The updated file is here: [https://github.com/HARPgroup/cbp6/blob/master/code/Modularized_Dashboard.Rmd](https://github.com/HARPgroup/cbp6/blob/master/code/Modularized_Dashboard.Rmd).  Due to the agnostic nature of these scripts, there are a large number of potential input variables to be defined -- the river segment, the data source for the first and second dataset (a string saying "vahydro", "gage", or "cbp_model"), the start and end dates, a string containing the location of the directory containing the auth.private file (which for us HARP-ers, is the directory containing the cbp6 github-linked directory), the site (http://deq2.bse.vt.edu/d.dh), whether model data should be pulled from the site or from the server (which is only an option if the dashboard is being generated on the server), and, depending on the data source specified earlier for the first and second datasets, the VA Hydro run IDs, gage number, or the model phase and scenario if pulling cbp6 model data from deq2.

If running it through Rstudio, there are two options for changing these inputs: either change them in the params header at the beginning of the document and click knit, or click the down arrow next to knit, select "knit with parameters", and change them in the pop-up window -- I only found out about this feature yesterday, and I think it's pretty cool.  Here's a pic of the pop-up:
![KnitWithParams](https://user-images.githubusercontent.com/36606075/68875706-455ee580-06d1-11ea-8dc2-4f4913875e11.PNG)

For now, the dashboard generator is still hard-wired to a directory containing the auth.private file -- it still reads in functions from the cbp6_functions.R and vahydro functions file as well, but we can update it to read from the cbp6 library when we update that library.

I'm trying to get this script up and running on the server at the moment -- I'll send out an update about that in the corresponding github issue when I have any success!
